source/setup.Rmd

```diff
@@ -1,282 +1,277 @@
-configuration for all pages.
+```{r docker-desktop-images, echo = FALSE, fig.cap = "The Docker Desktop images tab.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/docker-2.png") |> 
+  image_crop("3632x2000")
+```
 
-If your Mac computer has an Apple M1 processor chip you can download 
-the [latest Python 64-bit version from here](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh).
-After the download has finished, you need to run the downloaded script in the terminal using a command
-like:
+To start up a *container* using that image, click the play button beside the image. This will open the run configuration menu (Figure \@ref(fig:docker-desktop-runconfig)).
+In the "Host port" textbox, enter `8888`. In the "Volumes" section, click the "Host path" box and navigate to the folder where your Jupyter worksheets
+are stored. In the "Container path" text box, enter `/home/jovyan/work`. Then click the "Run" button to start the container.
 
-```
-bash path/to/Miniconda3-latest-MacOSX-arm64.sh
+```{r docker-desktop-runconfig, echo = FALSE, fig.cap = "The Docker Desktop container run configuration menu.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/docker-3.png") |> 
+  image_crop("3632x2000")
 ```
 
-Make sure to replace `path/to/` with the path of the folder
-containing the downloaded script. Most computers will save downloaded files to the `Downloads` folder. 
-If this is the case for your computer, you can run the script in the terminal by typing:
+After clicking the "Run" button, you will see a terminal. The terminal will then print 
+some text as the Docker container starts. Once the text stops scrolling, find the 
+URL in the terminal that starts 
+with `http://127.0.0.1:8888` (highlighted by the red box in Figure \@ref(fig:ubuntu-docker-terminal)), and paste it 
+into your browser to start JupyterLab.
 
-```
-bash Downloads/Miniconda3-latest-MacOSX-arm64.sh
+```{r docker-desktop-url, echo = FALSE, fig.cap = "The terminal text after running the Docker container in Ubuntu. The red box indicates the URL that you should paste into your browser to open JupyterLab.", fig.retina = 2, out.width="100%"}
+image_read("img/setup/docker-4.png") |> 
+  image_crop("3632x2000")
 ```
 
-The instructions for the installation will then appear. 
-Follow the prompts and agree to accepting the license,
-the default installation location,
-and to running `conda init`, which makes `conda` available from the terminal.
+## Working with JupyterLab Desktop
 
-**Ubuntu:** To install miniconda on Ubuntu, first download
-the [latest Python 64-bit version from here](https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh). 
-After the download has finished, open the terminal and execute the following 
-command:
+You can also run the worksheets accompanying this book on your computer
+using [JupyterLab Desktop](https://github.com/jupyterlab/jupyterlab-desktop).
```

Comment:
I make this suggestion as they are still "using" JupyterLab in Docker, the bigger difference here is that they are installing JupyterLab directly onto their OS.
```suggestion
by installing [JupyterLab Desktop](https://github.com/jupyterlab/jupyterlab-desktop).
```

Reply:
I think this just makes the sentence more awkward: "you can run X by installing Y" -- installing Y doesn't run X, it just sets things up to run X. Also JupyterLab Desktop is distinct from JupyterLab. I'm going to keep this one as-is I think.